general/v0.1/protocol_basics.md

```diff
@@ -37,40 +36,93 @@ Communication between fediverse server and FASP MUST use HTTPS in production
+`metadata` of their `nodeinfo` accessible via the `.well-known/nodeinfo`
+endpoint.
+
+
+Example `nodeinfo`:
+
+```json
+{
+  "version": "2.0",
+  "software": {
+    "name": "fediexample",
+    "version": "6.2.7"
+  },
+  "protocols": [
+    "activitypub"
+  ],
+  "services": {
+    "outbound": [],
+    "inbound": []
+  },
+  "openRegistrations": false,
+  "metadata": {
+    "nodeName": "fedi",
+    "faspBaseUrl": "https://fedi.example.com/fasp"
+  }
+}
+```
+
+### Request Integrity
+
+In order to allow both parties to verify the integrity of message
+contents, all requests MUST contain a `content-digest` HTTP header as
+defined by [RFC-9530](https://tools.ietf.org/html/rfc9530.html).
+
+The hashing algorith used MUST be SHA-256.
+
+### Authentication
 
 As described in [03: Registration](registration.md) both FASP and
-fediverse server use OAuth 2.0 to authorize API calls. Both MUST obtain a valid
-access token and send this as a "bearer token" in the `Authorization`
-HTTP header with every API call.
+fediverse server exchange client IDs and public keys. API requests are
+being authenticated by HTTP Message Signatures as defined in
+[RFC-9421](https://tools.ietf.org/html/rfc9421.html).
 
-Example header:
+The signature algorithm used is EdDSA Using Curve edwards25519.
+Signatures cover signature parameters, the derived components `@method`
+and `@target-uri` and the HTTP header `content-digest` (see previous
+section).
+
+The `keyid` parameter MUST include the client ID and the corresponding 
+private key is used to generate the signature.
+
+The required signature parameters are `created` and `keyid`.
+
+Example headers:
 
 ```http
-Authorization: Bearer SpBr6rheOp891mwWOfT6Pb"
+Content-Digest: sha-256=:RK/0qy18MlBSVnWgjwz6lZEWjP/lF5HF9bvEF8FabDg=:
+Signature-Input: sig1=("@method" "@target-uri" "content-digest"); created=1728467285;
+keyid="b2ks6vm8p23w"
+Signature: sig1=:+CcncFjyE+JAuwJO8MOEhRdyfShQz59e9bWDYGN3hoBorVp69k4V2PvS7zJiAoX3QchMlc47sUF4DsptUN+rDQ==:
 ```
 
-Both FASP and fediverse server MUST expire access tokens, forcing the other
-side to periodically request a new one.
-
-The OAuth 2.0 endpoint to request an access token MUST reside at the
-path `/oauth/token` that is relative to the base URL as described
-above. Existing fediverse software that already uses
-OAuth 2.0 and wants to add FASP support cannot re-use existing
-routes. This simplifies FASP implementation and
-enables fediverse software implementers to separate their existing OAuth
-2.0 implementation for regular API clients from the FASP API if so
-desired.
+The signature MUST be verified by the receiving party using the public
+key belonging to the transmitted client ID (`keyid` parameter).
+
+To ensure the integrity of the request, the derived components `@method`
+and `@target-uri` SHOULD be verified. Additionally the integrity of the
+`content-digest` HTTP header and its validity SHOULD be checked.
+
+It SHOULD be verified that the `created` timestamp is within an
```

Comment:
Ditto here, this seems like it should be a MUST

Reply:
See above.